src/only-export-components.test.ts

```diff
@@ -295,6 +300,11 @@ const invalid = [
     code: "export const MyComponent = () => {}; export const MyContext = React.createContext('test');",
     errorId: "reactContext",
   },
+  {
+    name: "should be invalid when custom HOC is used without adding it to the rule configuration",
+    code: "const MyComponent = () => {}; export default observer(MyComponent);",
+    errorId: ["localComponents", "anonymousExport"],
```

Comment:
I had to add support for multiple error IDs since we don't have a dedicated message for HOCs. I'm unsure how difficult it would be to detect HOCs while differentiating them from other types of exports and create a dedicated message ID